### PR TITLE
Storage cost calcullation

### PR DESCRIPTION
**Pull request by moldis**
- Storage cost calculation for premium allocation
- Unit-tests

### CODE REVIEW

File: zboxcore/sdk/allocation_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		err error
+	)
+
+	ssc.initialized = true
+	sdkInitialized = true
+	assert.NotNil(t, ssc.BlobberDetails)
+
+	minW, minR, err := ssc.GetMinWriteRead()
+	require.NoError(t, err)
+	assert.Equal(t, 800000/tokenUnit, minW)
```
Comment (Sriep) [inline zboxcore/sdk/allocation_test.go:24]: Why are you using both  `require` and `assert`? Can't you just stick to `require` and use EqualValue, InDelta or InEplsion etc... 
I think we want the test to stop as soon as there is an error.

**Review by Sriep — CHANGES_REQUESTED** [review]
Suggest only using require.

File: zboxcore/sdk/allocation_test.go (hunk 1 above)
Comment (moldis) [reply zboxcore/sdk/allocation_test.go]: Resolved

**Comment by Sriep** [comment]
Needs to fix merge conflicts.

**Review by avanaur — COMMENTED** [review]
We have this calculation of storage (write/upload) on zboxcli and seems not same in terms of formula. 

I think what's own zboxcli is correct since it factored in the number of data and parity shards. Can you confirm @guruhubb?

File: zboxcore/sdk/allocation_test.go [hunk 2]
```diff
@@ -1,12 +1,126 @@
...
 	"github.com/0chain/gosdk/zboxcore/blockchain"
 	"github.com/0chain/gosdk/zboxcore/fileref"
+	"github.com/stretchr/testify/require"
 	"os"
 	"testing"
 )
 
+const tokenUnit = 10000000000.0
+
+func Test_getMinWriteRead_success(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (guruhubb) [reply zboxcore/sdk/allocation_test.go:14]: @moldis have you looked at download and upload cost in zboxcli?

File: zboxcore/sdk/allocation.go [hunk 3]
```diff
@@ -1144,3 +1145,65 @@ func (a *Allocation) RemoveCollaborator(filePath, collaboratorID string) error {
...
+	if !a.isInitialized(){
+		return 0,0, notInitialized
+	}
+
+	blobbersCopy := a.BlobberDetails
+	if len(blobbersCopy) == 0 {
+		return 0,0, noBLOBBERS
+	}
+
+	sort.Slice(blobbersCopy, func(i, j int) bool {
```
[the remark this answers is not in the document]
Comment (moldis) [reply zboxcore/sdk/allocation.go]: Fixed

File: zboxcore/sdk/allocation_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (moldis) [reply zboxcore/sdk/allocation_test.go]: Fixed. Just want to confirm cost I am looking for is on this page: https://one.devnet-0chain.net/0box/dashboard/allocateStorage 
If it's upload cost it's in this branch. Consider to move calculation in single place, as for now it's 3 different places with seprated code (cli, sdk and explorer)
сс @avanaur @guruhubb

[the remark this answers is not in the document]
Comment (guruhubb) [reply zboxcore/sdk/allocation_test.go]: Ok go ahead

**Pull request merged**